### PR TITLE
fix(amplify-provider-awscloudformation): batch cfn status polling

### PR DESCRIPTION
**Pull request by yuth**
Cloud formation provider polls cloud formation to get the list of events to show when pushing the
stack. When a project has more than 20 resources, this causes throtlling. Fixing this issue by
batching the polling and retrying

*Issue #, if available:*
#705
*Description of changes:*

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/amplify-provider-awscloudformation/src/aws-utils/aws-cfn.js [hunk 1]
```diff
@@ -75,106 +88,102 @@ class CloudFormation {
...
-                // Just get the new events to display to the user
-
-                newEvents = _.differenceBy(newEvents, allShownEvents, 'EventId');
-                newEvents = _.uniqBy(newEvents, 'EventId');
-                showEvents(newEvents);
-                allShownEvents = allShownEvents.concat(newEvents);
-              });
           }
 
           let newEvents = stackEvents;
```
Comment (kaustavghosh06) [inline packages/amplify-provider-awscloudformation/src/aws-utils/aws-cfn.js]: Shouldn't there be an else out here?

File: packages/amplify-provider-awscloudformation/src/aws-utils/aws-cfn.js [hunk 2]
```diff
@@ -75,106 +88,102 @@ class CloudFormation {
...
-        nestedstackEvents = nestedstackEvents.filter(event =>
-          self.eventStartTime < new Date(event.Timestamp));
+        nestedstackEvents = nestedstackEvents
+          .filter(event => self.eventStartTime < new Date(event.Timestamp));
 
         return nestedstackEvents;
-      });
+      })
+      .catch(() =>
+        // ignore. We dont fail if we can't get the nested stack status
```
Comment (kaustavghosh06) [inline packages/amplify-provider-awscloudformation/src/aws-utils/aws-cfn.js]: I think we should have a check for the CFN errors. There might be other errors as well which will go uncaught and hard to debug in a production environment.

Comment (yuth) [reply packages/amplify-provider-awscloudformation/src/aws-utils/aws-cfn.js]: OK

File: packages/amplify-provider-awscloudformation/src/aws-utils/aws-cfn.js [hunk 3]
```diff
@@ -7,22 +7,32 @@ const providerName = require('../../lib/constants').ProviderName;
...
 
+const CFN_MAX_CONCURRENT_REQUEST = 15;
+const CFN_POLL_TIME = 7000; // 7sec
+
+const CFN_SUCCESS_STATUS = [
+  'UPDATE_COMPLETE',
+  'CREATE_COMPLETE',
+  'DELETE_COMPLETE',
+  'DELETE_SKIPPED',
+];
```
Comment (UnleashedMind) [inline packages/amplify-provider-awscloudformation/src/aws-utils/aws-cfn.js]: Maybe rename it TERMINAL_STATUS and add all the statuses that do not need further checking?

File: packages/amplify-provider-awscloudformation/src/aws-utils/aws-cfn.js (hunk 1 above)
Comment (yuth) [reply packages/amplify-provider-awscloudformation/src/aws-utils/aws-cfn.js]: We don't need else block as we dedupe the events when showing

**Pull request merged**